### ISSUE TITLE
[UselessParenthesesSniff] False positive with generator

### ISSUE THREAD

**Issue by enumag** [slevomat/coding-standard#618]
(Using latest 4.x release, can't test 5.0 yet.)

Currently the sniff changes this code:

```php
$state = (yield from self::removeContacts($state, $command->contacts)) ?? $state;
```

to this:

```php
$state = yield from self::removeContacts($state, $command->contacts) ?? $state;
```

This is wrong because `??` apparently has higher precedence than `yield from` so the fixed code behaves the same as

```php
$state = yield from (self::removeContacts($state, $command->contacts) ?? $state);
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/2c0ecbcdbc2c1f78724772381587803d12429c44

**Comment by enumag** [comment]
@kukulich You might want to add `yield` as well. It has nearly the same operator priority as `yield from` so it most likely suffers from the same mistake.
